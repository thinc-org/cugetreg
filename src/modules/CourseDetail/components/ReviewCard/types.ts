import { Review } from '@/common/types/reviews'

export interface ReviewCardProps extends Review {}
