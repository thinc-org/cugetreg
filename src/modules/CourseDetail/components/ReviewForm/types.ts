export interface ReviewFormProps {
  courseNo: string
}
