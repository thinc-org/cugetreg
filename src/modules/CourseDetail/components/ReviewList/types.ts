import { Course } from '@thinc-org/chula-courses'

export interface ReviewListProps {
  course: Course
}
