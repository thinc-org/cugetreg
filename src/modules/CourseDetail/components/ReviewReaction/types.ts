import { ReviewInteraction } from '@/common/types/reviews'

export interface ReviewReactionProps {
  type: ReviewInteraction
  defaultPressed: boolean
  reactionCount: number
}
