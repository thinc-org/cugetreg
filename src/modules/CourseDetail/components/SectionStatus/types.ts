import { TypographyProps } from '@material-ui/core'
import { Capacity } from '@thinc-org/chula-courses'

export type AvailableStatus = 'avialable' | 'full' | 'closed'
export interface SectionStatusProps extends TypographyProps {
  capacity: Capacity
  closed: boolean
}
