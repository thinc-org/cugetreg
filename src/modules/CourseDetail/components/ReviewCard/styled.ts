import isPropValid from '@emotion/is-prop-valid'
import styled from '@emotion/styled'
import { Typography, Stack, StackProps } from '@mui/material'

import { RichTextRenderer } from '@/modules/RichText'

export const Card = styled(Stack, { shouldForwardProp: isPropValid })<StackProps & { pending?: boolean }>`
  border-radius: ${({ theme }) => theme.shape.borderRadius}px;
  border: 1px solid ${({ theme }) => theme.palette.divider};
  padding: ${({ theme }) => theme.spacing(4)};
  ${({ theme }) => theme.breakpoints.down('sm')} {
    padding: ${({ theme }) => theme.spacing(2)};
  }
  opacity: ${({ pending }) => (pending ? 0.6 : 1)};
`

const MainTypography = styled(Typography)`
  color: ${({ theme }) => theme.palette.primary.main};
`
export const CardTerm = styled(MainTypography)``
CardTerm.defaultProps = { variant: 'h5' }

export const CardContent = styled(RichTextRenderer)``

export const CardRating = styled(MainTypography)`
  align-self: flex-end;
`
CardRating.defaultProps = { variant: 'h5' }

export const CardMaxRating = styled(Typography)`
  align-self: flex-end;
  line-height: 24px;
  color: ${({ theme }) => theme.palette.primaryRange[100]};
  ${({ theme }) => theme.breakpoints.down('sm')} {
    display: none;
  }
`
CardMaxRating.defaultProps = { variant: 'subtitle2' }

export const CardRejectedMessage = styled(Typography)`
  color: ${({ theme }) => theme.palette.highlight.red[500]};
`
CardRejectedMessage.defaultProps = { variant: 'subtitle2' }
