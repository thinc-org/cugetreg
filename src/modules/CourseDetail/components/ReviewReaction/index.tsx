import { Stack, Typography } from '@mui/material'
import { useTheme } from '@mui/system'
import { IconBaseProps } from 'react-icons/lib'
import { MdThumbUpOffAlt, MdThumbUp, MdThumbDownOffAlt, MdThumbDown } from 'react-icons/md'

import { ReviewInteraction } from '@/common/types/reviews'

import { ReviewReactionProps } from './types'

export const ReviewReaction: React.FC<ReviewReactionProps> = ({ type, pressed, reactionCount, onClick }) => {
  const theme = useTheme()

  const getColor = () => {
    if (pressed) {
      switch (type) {
        case ReviewInteraction.Like:
          return theme.palette.highlight.green[500]
        case ReviewInteraction.Dislike:
          return theme.palette.highlight.red[500]
      }
    }
    return theme.palette.primaryRange[50]
  }

  const SelectedIcon = (props: IconBaseProps) => {
    const color = theme.palette.primaryRange[50]
    const modifiedProps: IconBaseProps = { ...props, color, size: 20 }
    switch (type) {
      case ReviewInteraction.Like:
        return !pressed ? <MdThumbUpOffAlt {...modifiedProps} /> : <MdThumbUp {...modifiedProps} color={getColor()} />
      case ReviewInteraction.Dislike:
        return !pressed ? (
          <MdThumbDownOffAlt {...modifiedProps} />
        ) : (
          <MdThumbDown {...modifiedProps} color={getColor()} />
        )
      default:
        return !pressed ? <MdThumbUpOffAlt {...modifiedProps} /> : <MdThumbUp {...modifiedProps} color={getColor()} />
    }
  }

  return (
    <Stack spacing={1} direction="row" alignItems="center" onClick={onClick}>
      <SelectedIcon />
      <Typography variant="h6" color={getColor()}>
        {reactionCount}
      </Typography>
    </Stack>
  )
}
