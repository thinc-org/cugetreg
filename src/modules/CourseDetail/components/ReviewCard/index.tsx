import { IconButton, Stack, useTheme } from '@mui/material'
import { MdFlag, MdDeleteOutline, MdOutlineStar, MdEdit } from 'react-icons/md'

import { GeneralChip } from '@/common/components/Chips'
import { OtherChipKey } from '@/common/components/Chips/config'
import { ReviewInteractionType, ReviewStatus } from '@/common/types/reviews'
import { getSemesterName } from '@/common/utils/getSemesterName'
import { useReviewContext } from '@/modules/CourseDetail/context/Review'

import { ReviewReaction } from '../ReviewReaction'
import { Card, CardTerm, CardContent, CardRating, CardMaxRating, CardRejectedMessage } from './styled'
import { ReviewCardProps } from './types'

export const ReviewCard: React.FC<ReviewCardProps> = (data) => {
  const theme = useTheme()
  const actionIconProps = {
    size: 24,
    color: theme.palette.primaryRange[50],
  }

  const term = `${data.academicYear} ${getSemesterName(data.semester)}`

  const getChipType = (): OtherChipKey | null => {
    switch (data.status) {
      case ReviewStatus.Pending:
        return 'reviewPending'
      case ReviewStatus.Rejected:
        return 'reviewRejected'
      default:
        return null
    }
  }

  const chipType = getChipType()

  const { setInteraction, reportReview, deleteMyReview, editMyReview } = useReviewContext()

  const handleLikeClick = () => {
    setInteraction(data._id, ReviewInteractionType.Like)
  }

  const handleDislikeClick = () => {
    setInteraction(data._id, ReviewInteractionType.Dislike)
  }

  const handleReportClick = () => {
    reportReview(data._id)
  }

  const handleDeleteClick = () => {
    deleteMyReview(data._id)
  }

  const handleEditClick = () => {
    const srollToElement = document.getElementById('review-title')
    if (srollToElement) {
      const offset = document.documentElement.clientHeight * 0.35
      const offsetTop = srollToElement.offsetTop
      window.scrollTo({ top: offsetTop - offset, behavior: 'smooth' })
    }
    editMyReview(data._id)
  }

  return (
    <Card direction="column" spacing={2}>
      <Stack direction="row" justifyContent="space-between" gap={1} flexWrap="wrap">
        <Stack direction="row" spacing={2} alignItems="center">
          <CardTerm>{term}</CardTerm>
          {chipType && <GeneralChip type={chipType} />}
        </Stack>
        <Stack direction="row" spacing={1} alignItems="center">
          <MdOutlineStar size={20} />
          <CardRating>{data.rating / 2}</CardRating>
          <CardMaxRating>จาก 5</CardMaxRating>
        </Stack>
      </Stack>
      <CardContent>{data.content}</CardContent>

      <Stack direction="row" justifyContent="space-between" alignItems="center">
        {/** Left side */}
        {data.status === ReviewStatus.Rejected ? (
          <CardRejectedMessage>กรุณาแก้ไข ก่อนส่งความคิดเห็นอีกครั้ง</CardRejectedMessage>
        ) : (
          <Stack direction="row" spacing={3}>
            <ReviewReaction
              type={ReviewInteractionType.Like}
              pressed={data.myInteraction === ReviewInteractionType.Like}
              reactionCount={data.likeCount}
              onClick={handleLikeClick}
            />
            <ReviewReaction
              type={ReviewInteractionType.Dislike}
              pressed={data.myInteraction === ReviewInteractionType.Dislike}
              reactionCount={data.dislikeCount}
              onClick={handleDislikeClick}
            />
          </Stack>
        )}

        {/** Right side */}
        {data.isOwner ? (
          <Stack direction="row" spacing={1} ml="auto">
            <IconButton size="small" onClick={handleEditClick}>
              <MdEdit {...actionIconProps} />
            </IconButton>
            <IconButton size="small" onClick={handleDeleteClick}>
              <MdDeleteOutline {...actionIconProps} />
            </IconButton>
          </Stack>
        ) : (
          <IconButton size="small" onClick={handleReportClick}>
            <MdFlag {...actionIconProps} />
          </IconButton>
        )}
      </Stack>
    </Card>
  )
}
