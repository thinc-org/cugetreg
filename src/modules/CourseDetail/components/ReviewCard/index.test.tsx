import { ThemeProvider } from '@mui/material'
import { shallow } from 'enzyme'
import { MdDeleteOutline, MdEdit, MdFlag } from 'react-icons/md'

import { mockMyPendingReviews, mockMyRejectedReviews, mockMyReviews, mockReviews } from '@/__mock__/review'
import { ReviewInteractionType } from '@/common/types/reviews'
import { lightTheme } from '@/configs/theme'
import { ReviewCardProps } from '@/modules/CourseDetail/components/ReviewCard/types'
import { ReviewReaction } from '@/modules/CourseDetail/components/ReviewReaction'

describe('ReviewCard', () => {
  const mockReviewData = mockReviews[0]
  const mockMyReviewData = mockMyReviews[0]
  const mockMyPendingReviewData = mockMyPendingReviews[0]
  const mockMyRejectedReviewData = mockMyRejectedReviews[0]

  const useThemeSpy = jest.fn(() => lightTheme)
  const setInteractionSpy = jest.fn()
  const reportReviewSpy = jest.fn()
  const deleteMyReviewSpy = jest.fn()
  const editMyReviewSpy = jest.fn()
  const useReviewContextSpy = jest.fn(() => ({
    setInteraction: setInteractionSpy,
    reportReview: reportReviewSpy,
    deleteMyReview: deleteMyReviewSpy,
    editMyReview: editMyReviewSpy,
  }))

  jest.doMock('@/modules/CourseDetail/context/Review', () => ({
    useReviewContext: useReviewContextSpy,
  }))

  jest.doMock('@mui/material', () => ({
    ...jest.requireActual('@mui/material'),
    useTheme: useThemeSpy,
  }))

  afterEach(() => {
    jest.clearAllMocks()
  })

  async function customShallow(props: ReviewCardProps) {
    const { ReviewCard } = await import('.')
    return shallow(<ReviewCard {...props} />, {
      wrappingComponent: ThemeProvider,
      wrappingComponentProps: { theme: lightTheme },
    })
  }

  it('should render correctly if this is not your review', async () => {
    const wrapper = await customShallow(mockReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your approved review', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your pending review', async () => {
    const wrapper = await customShallow(mockMyPendingReviewData)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render correctly if this is your rejected review', async () => {
    const wrapper = await customShallow(mockMyRejectedReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('should not have edit icon and delete icon but report icon if this is not your review', async () => {
    const wrapper = await customShallow(mockReviewData)
    expect(wrapper.find(MdDeleteOutline)).toHaveLength(0)
    expect(wrapper.find(MdEdit)).toHaveLength(0)
    expect(wrapper.find(MdFlag)).toHaveLength(1)
  })

  it('should not have report icon but edit icon and delete icon but have report icon if this is your review', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    expect(wrapper.find(MdDeleteOutline)).toHaveLength(1)
    expect(wrapper.find(MdEdit)).toHaveLength(1)
    expect(wrapper.find(MdFlag)).toHaveLength(0)
  })

  it('should not have rejected message if this is not a rejected review', async () => {
    let wrapper = await customShallow(mockReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
    wrapper = await customShallow(mockMyReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
    wrapper = await customShallow(mockMyPendingReviewData)
    expect(wrapper.text().includes('กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง')).toBeFalsy()
  })

  it('should call setInteraction with like interaction if we click like', async () => {
    const wrapper = await customShallow(mockReviewData)
    const likeButton = wrapper.find(ReviewReaction).at(0)
    likeButton.simulate('click')
    expect(setInteractionSpy).toHaveBeenCalledTimes(1)
    expect(setInteractionSpy).toHaveBeenCalledWith(mockReviewData._id, ReviewInteractionType.Like)
  })

  it('should call setInteraction with dislike interaction if we click dislike', async () => {
    const wrapper = await customShallow(mockReviewData)
    const dislikeButton = wrapper.find(ReviewReaction).at(1)
    dislikeButton.simulate('click')
    expect(setInteractionSpy).toHaveBeenCalledTimes(1)
    expect(setInteractionSpy).toHaveBeenCalledWith(mockReviewData._id, ReviewInteractionType.Dislike)
  })

  it('should call deleteMyReview when click delete icon', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    const deleteIcon = wrapper.find(MdDeleteOutline).at(0).parent()
    deleteIcon.simulate('click')
    expect(deleteMyReviewSpy).toHaveBeenCalledTimes(1)
    expect(deleteMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  })

  it('should call reportReview when click report icon', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    const reportIcon = wrapper.find(MdDeleteOutline).at(0).parent()
    reportIcon.simulate('click')
    expect(deleteMyReviewSpy).toHaveBeenCalledTimes(1)
    expect(deleteMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  })

  it('should call editReview when click edit icon', async () => {
    const wrapper = await customShallow(mockMyReviewData)
    const editIcon = wrapper.find(MdEdit).at(0).parent()
    editIcon.simulate('click')
    expect(editMyReviewSpy).toHaveBeenCalledTimes(1)
    expect(editMyReviewSpy).toHaveBeenCalledWith(mockMyReviewData._id)
  })
})
