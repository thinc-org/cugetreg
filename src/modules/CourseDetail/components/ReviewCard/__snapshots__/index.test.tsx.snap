// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard should render correctly if this is not your review 1`] = `
<Card
  direction="column"
  pending={false}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <CardTerm
        variant="h5"
      >
        2564 ภาคปลาย
      </CardTerm>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <CardRating
        variant="h5"
      >
        4
      </CardRating>
      <CardMaxRating
        variant="subtitle2"
      >
        จาก 5
      </CardMaxRating>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <HighlightHTML
    dangerouslySetInnerHTML={
      Object {
        "__html": "เหนื่อยากเลยค่าแงงงงง",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Card>
`;

exports[`ReviewCard should render correctly if this is your approved review 1`] = `
<Card
  direction="column"
  pending={false}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <CardTerm
        variant="h5"
      >
        2564 ภาคปลาย
      </CardTerm>
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <CardRating
        variant="h5"
      >
        4
      </CardRating>
      <CardMaxRating
        variant="subtitle2"
      >
        จาก 5
      </CardMaxRating>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <HighlightHTML
    dangerouslySetInnerHTML={
      Object {
        "__html": "เหนื่อยากเลยค่าแงงงงง",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Card>
`;

exports[`ReviewCard should render correctly if this is your pending review 1`] = `
<Card
  direction="column"
  pending={true}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <CardTerm
        variant="h5"
      >
        2564 ภาคปลาย
      </CardTerm>
      <GeneralChip
        type="reviewPending"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <CardRating
        variant="h5"
      >
        1.5
      </CardRating>
      <CardMaxRating
        variant="subtitle2"
      >
        จาก 5
      </CardMaxRating>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <HighlightHTML
    dangerouslySetInnerHTML={
      Object {
        "__html": "เขียนดี ๆ แล้ว อิอิ",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      direction="row"
      spacing={3}
    >
      <ReviewReaction
        onClick={[Function]}
        pressed={true}
        reactionCount={1}
        type="L"
      />
      <ReviewReaction
        onClick={[Function]}
        pressed={false}
        reactionCount={0}
        type="D"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Card>
`;

exports[`ReviewCard should render correctly if this is your rejected review 1`] = `
<Card
  direction="column"
  pending={true}
  spacing={2}
>
  <ForwardRef(Stack)
    direction="row"
    flexWrap="wrap"
    gap={1}
    justifyContent="space-between"
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={2}
    >
      <CardTerm
        variant="h5"
      >
        2564 ภาคปลาย
      </CardTerm>
      <GeneralChip
        type="reviewRejected"
      />
    </ForwardRef(Stack)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      spacing={1}
    >
      <MdOutlineStar
        size={20}
      />
      <CardRating
        variant="h5"
      >
        1.5
      </CardRating>
      <CardMaxRating
        variant="subtitle2"
      >
        จาก 5
      </CardMaxRating>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
  <HighlightHTML
    dangerouslySetInnerHTML={
      Object {
        "__html": "เขียนดี ๆ แล้ว อิอิ",
      }
    }
  />
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    justifyContent="space-between"
  >
    <CardRejectedMessage
      variant="subtitle2"
    >
      กรุณาแก้ไข ก่อนส่งรีวิวรายวิชาอีกครั้ง
    </CardRejectedMessage>
    <ForwardRef(Stack)
      direction="row"
      ml="auto"
      spacing={1}
    >
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdEdit
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
      <ForwardRef(IconButton)
        onClick={[Function]}
        size="small"
      >
        <MdDelete
          color="#C9CCE4"
          size={24}
        />
      </ForwardRef(IconButton)>
    </ForwardRef(Stack)>
  </ForwardRef(Stack)>
</Card>
`;
